llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -5468,7 +5467,8 @@ Instruction *InstCombinerImpl::foldICmpBinOp(ICmpInst &I,
         break;
 
       const APInt *C;
-      if (match(BO0->getOperand(1), m_APInt(C)) && !C->isZero() &&
+      if (BO0->hasOneUse() && BO1->hasOneUse() &&
```

Comment:
Should it be ||?

Reply:
No. We create two `and` instructions and one `icmp` here. See the test `eq_mul_constants_with_tz_extra_use1`.

Reply:
Yeah fair enough, just this case is not relaxing the constraint, its tightening it.

Reply:
Yes. I already pointed out this in the PR description.